docs/lib/datastore/fragments/ios/auth-model.md

```diff
@@ -0,0 +1,104 @@
+<amplify-callout>
+When using the `@auth` directive with DataStore, currently only following fields are supported:
+
+1. `allow` with "owner" and "groups"
+2. `ownerField`
+3. `groups`
+4. `operations`
+
+</amplify-callout>
+
+## Updated schema
+
+Create a Model that is read-only.
+
+```graphql
+type Todo
+    @model
+    @auth(rules: [
+        { allow: owner, ownerField: "owner", operations: [create, update, delete] },
+    ]) {
+    id: ID!
+    content: String!
+    owner: String
+}
+```
+
+The API will allow users to query, list, and subscribe to model creation, updates, deletions by other users. A user can only update and delete their own model, that is a model can not be updated or deleted by another user.
+
+## API with Cognito User Pool auth
+
+With the schema, make sure you have selected Cognito User Pool as the auth
+
+```console
+? Please select from one of the below mentioned services: 
+  `GraphQL`
+? Select from the options below 
+  `Update auth settings`
+? Choose the default authorization type for the API 
+  `Amazon Cognito User Pool`
+```
+
+Once you have updated the schema and selected Cognito User Pool as the authorization type, run `amplify push` 
+
+## Configuration
+
+`amplifyconfiguration.json` will have section under "awsAPIPlugins" and "awsCoognitoAuthPlugin" with the API and Auth configurations
+```json
+{
+    "api": {
+        "plugins": {
+            "awsAPIPlugin": { 
+                // ...
+    "auth": {
+        "plugins": {
+            "awsCognitoAuthPlugin": {
+                // ...
+```
+
+Add "AWSCognitoAuthPlugin" and "AWSAPIPlugin" to Amplify before configuring
+
+```swift
+Amplify.add(plugin: AWSCognitoAuthPlugin())
+Amplify.add(plugin: AWSAPIPlugin())
+```
+
+## Saving
+
+In order to save models, make sure the user is signed in using `Amplify.Auth.signIn`, then you can create an instance of the model as normal and call `DataStore.save`.
+
+If the user's session has expired or the user has logged out but the save is still executed, the model instance will be persisted to local storage. The model instance however will fail to be saved to API and the error handler will be called. See [conflict resolution section](~/lib/datastore/conflict.md) to learn more about how to handle errors in the DataStoreConfiguration's `errorhandler`. To prevent unnecessary saves from occurring, you can check if the user is signed in using `Amplify.Auth.fetchAuthSession`:
```

Comment:
This information is good, but i think needs to be moved to a section (or another page?) where we explain more than just this...